Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/AppConfigProvider.kt

```diff
@@ -0,0 +1,106 @@
+package de.rki.coronawarnapp.appconfig
+
+import androidx.annotation.VisibleForTesting
+import dagger.Lazy
+import de.rki.coronawarnapp.diagnosiskeys.server.LocationCode
+import de.rki.coronawarnapp.environment.download.DownloadCDNHomeCountry
+import de.rki.coronawarnapp.server.protocols.ApplicationConfigurationOuterClass
+import de.rki.coronawarnapp.util.ZipHelper.unzip
+import de.rki.coronawarnapp.util.security.VerificationKeys
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class AppConfigProvider @Inject constructor(
+    private val appConfigAPI: Lazy<AppConfigApiV1>,
+    private val verificationKeys: VerificationKeys,
+    @DownloadCDNHomeCountry private val homeCountry: LocationCode,
+    private val configStorage: AppConfigStorage
+) {
+
+    private val configApi: AppConfigApiV1
+        get() = appConfigAPI.get()
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal suspend fun downloadAppConfig(): ByteArray? {
+        Timber.tag(TAG).d("Fetching app config.")
+        var exportBinary: ByteArray? = null
+        var exportSignature: ByteArray? = null
+        configApi.getApplicationConfiguration(homeCountry.identifier).byteStream()
```

Comment:
do we need to call .use() on the byteStream here to close it properly?

Reply:
Hm don't think so, the `unzip` should close the stream, it internally uses `use {}` on the bytestream.